PEP 686: Grammar fixes (#2464)
python/peps: pep-0686.rst

Abstract
========

> This PEP proposes to enable :pep:`UTF-8 mode <540>` by default.

Reviewer: ```suggestion
This PEP proposes enabling :pep:`UTF-8 mode <540>` by default.
```

With this change, Python consistently uses UTF-8 for default encoding of
files, stdio, and pipes.

Motivation
==========

UTF-8 becomes de facto standard text encoding.

> * Default encoding of Python source files is UTF-8.

Reviewer: ```suggestion
* The default encoding of Python source files is UTF-8.
```

* JSON, TOML, YAML use UTF-8.

> * Most text editors, including VS Code and Windows notepad use UTF-8 by

Reviewer: ```suggestion
* Most text editors, including Visual Studio Code and Windows Notepad use UTF-8 by
```

  default.
* Most websites and text data on the internet use UTF-8.

> * And many other popular programming languages, including node.js, Go, Rust,

Reviewer: ```suggestion
* And many other popular programming languages, including Node.js, Go, Rust,
```

  and Java uses UTF-8 by default.

> Changing the default encoding to UTF-8 makes Python easier to interoperate

Reviewer: ```suggestion
Changing the default encoding to UTF-8 makes it easier for Python to interoperate
```

with them.

Additionally, many Python developers using Unix forget that the default

> encoding is a platform dependent.

Reviewer: ```suggestion
encoding is platform dependent.
```

``TextIOWrapper`` should use locale encoding when ``encoding="locale"`` is
passed before/after the default encoding is changed to UTF-8.

To fix this inconsistency, we will add ``locale.get_encoding()``.
It is the same as ``locale.getpreferredencoding(False)`` but it ignores
the UTF-8 mode.

This change will be released in Python 3.11 so that users can use UTF-8 mode
that is the same as Python 3.13.

Backward Compatibility
======================

Most Unix systems use UTF-8 locale and Python enables UTF-8 mode when its
locale is C or POSIX.
So this change mostly affects Windows users.

When a Python program depends on the default encoding, this change may cause
``UnicodeError``, mojibake, or even silent data corruption.
So this change should be announced loudly.

To resolve this backward incompatibility, users can do:

* Disable UTF-8 mode.

> * Use ``EncodingWarning`` to find where the default encoding used and use

Reviewer: ```suggestion
* Use ``EncodingWarning`` to find where the default encoding is used and use
```

Deprecate implicit encoding
---------------------------

Deprecating the use of the default encoding is considered.

But there are many cases that the default encoding is used for reading/writing
only ASCII text.
Additionally, such warnings are not useful for non-cross platform applications
run on Unix.

So forcing users to specify the ``encoding`` everywhere is too painful.

> Java also rejected this idea in the `JEP 400`_.

Reviewer: ```suggestion
Java also rejected this idea in `JEP 400`_.
```
